cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -255,15 +316,19 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 		if seen[uint64(start)] {
 			continue
 		}
+
+		// Check if this partition should be encrypted
+		if encryptData && s.Role == gadget.SystemData {
+			p.Type = "e8,CA7D7CCB-63ED-4C53-861C-1742536059CC"
```

Comment:
p is a copy, so the writing in place is ok. But I would still move the partitionType call earlier out of the args and avoid that

Reply:
I kept the partitionType call as is but assigned the type to a different variable instead of writing on the structure. (Very explicit, but if we adopt this practice we'll never slip and end up modifying something passed as a pointer)

Reply:
Oh, now I see why I replaced it in place. The structure copy is used in the list of partitions to be created that buildPartitionList() returns, and to be consistent with what is actually created the partition ID should be set to the encrypted partition ID if this is the case. I'll add a comment to explain that.

Reply:
Something is off here. `LaidOutStructure` embeds a `*VolumeStructure` and p.Type is part of VoulmeStructure. Perhaps this should be updated explicitly in `Run()`, before `CreateMissing` is called?

Reply:
I was confused here, I thought p was a copy because I confused types, anyway if it's *  we should really not modify it in place this way

Reply:
Run seems the wrong place for that too though

Reply:
Discussed further on IRC. The agreement is to drop the type change for now. From what we understand, using the right type does not buy us anything in particular, but causes some issues with handling this in code cleanly.